### PR TITLE
feat(segment-view): adds support for new `ion-segment-view` component

### PR DESCRIPTION
**Pull request by tanner-reits**
Issue number: resolves internal

---------

## What is the current behavior?

Segments can only be changed by clicking a segment button, or dragging the indicator

## What is the new behavior?

The segment/segment buttons can now be linked to segment content within a segment view component. This content is scrollable/swipeable. Changing the content will update the segment/indicator and vice-versa.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

**Limitations:**
- Segment buttons **cannot** be disabled when connected ton `ion-segment-content` instances
- The `ion-segment` **cannot** be without a value when linked with an `ion-segment-view`. If no value is provided, the value will default to the value of the first `ion-segment-content`

[Preview](https://ionic-framework-jlt8by2io-ionic1.vercel.app/src/components/segment-view/test/basic)
[Preview (disabled state)](https://ionic-framework-jlt8by2io-ionic1.vercel.app/src/components/segment-view/test/disabled)

### CODE REVIEW

File: core/src/components/segment-button/segment-button.tsx [hunk 1]
```diff
@@ -36,10 +36,15 @@ export class SegmentButton implements ComponentInterface, ButtonInterface {
...
+  /**
+   * The `id` of the segment content.
+   */
+  @Prop({ reflect: true }) contentId?: string;
+
   /**
    * If `true`, the user cannot interact with the segment button.
    */
-  @Prop({ mutable: true }) disabled = false;
+  @Prop({ mutable: true, reflect: true }) disabled = false;
```
Comment (tanner-reits) [inline core/src/components/segment-button/segment-button.tsx:47]: Not sure why this wasn't reflected before, but I can revert if we don't want this. It was really just to make testing the behavior of buttons not being allowed to be disabled with a segment view easier

File: core/src/components/segment-view/test/basic/segment-view.e2e.ts [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+        <ion-segment-view>
+          <ion-segment-content id="paid">Paid</ion-segment-content>
+          <ion-segment-content id="free">Free</ion-segment-content>
+          <ion-segment-content id="top">Top</ion-segment-content>
+        </ion-segment-view>
+      `,
+        config
+      );
+
+      await page.click('ion-segment-button[value="top"]');
```
Comment (thetaPC) [inline core/src/components/segment-view/test/basic/segment-view.e2e.ts:86]:
```suggestion
      await page.locator.click('ion-segment-button[value="top"]');
```

`page.click` is [deprecated](https://playwright.dev/docs/api/class-page#page-click) and should be using `locator.click` instead.

**Review by thetaPC — COMMENTED** [review]
Issues have been fixed. Minor changes are requested.

File: core/src/components/segment-view/segment-view.tsx [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+   * Emitted when the segment view is scrolled.
+   */
+  @Event() ionSegmentViewScroll!: EventEmitter<SegmentViewScrollEvent>;
+
+  /**
+   * Emitted when the segment view scroll has ended.
+   */
+  @Event() ionSegmentViewScrollEnd!: EventEmitter<void>;
+
+  @Event() ionSegmentViewScrollStart!: EventEmitter<void>;
```
[the remark this answers is not in the document]
Comment (tanner-reits) [reply core/src/components/segment-view/segment-view.tsx]: We don't actually use these anywhere and I don't suspect they're very helpful for users either, so I just went ahead and removed them

**Review by thetaPC — APPROVED** [review]
LGTM

File: core/src/components/segment-button/segment-button.tsx (hunk 1 above)
Comment (brandyscarney) [reply core/src/components/segment-button/segment-button.tsx:47]: I don't see a reason for reflecting it if it works fine without it.

**Review by brandyscarney — APPROVED** [review]

File: core/src/components/segment-button/segment-button.tsx (hunk 1 above)
Comment (tanner-reits) [reply core/src/components/segment-button/segment-button.tsx]: Reverted :)

**Pull request merged**